extensions/cdc-core/src/main/java/com/hazelcast/jet/cdc/ChangeEventElement.java

```diff
@@ -0,0 +1,189 @@
+/*
+ * Copyright 2020 Hazelcast Inc.
+ *
+ * Licensed under the Hazelcast Community License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://hazelcast.com/hazelcast-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.cdc;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.hazelcast.jet.annotation.EvolvingApi;
+
+import javax.annotation.Nonnull;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Arbitrary part of a {@link ChangeEvent}, as big as the whole body or
+ * as small as a single patch expression, based on a complete JSON
+ * expression. Contains various methods for retrieving component values
+ * or for mapping itself to data objects.
+ *
+ * @since 4.1
+ */
+@EvolvingApi
+public interface ChangeEventElement {
+
+    /**
+     * Maps the entire element to an instance of the specified class.
+     * <p>
+     * For databases providing standard JSON syntax, parsing it is based
+     * on <a
+     * href="https://github.com/FasterXML/jackson-databind">Jackson
+     * Databind</a>, in particular on the Jackson {@code ObjectMapper},
+     * so the parameter class needs to be annotated accordingly.
+     *
+     * @return optional {@code Object} value, which is empty only if the
+     * specified key is not found or if it's value is null
+     * @throws ParsingException if the whole structure containing this
+     *                          element is unparsable or the mapping
+     *                          fails to produce a result
+     */
+    @Nonnull
+    <T> T mapToObj(Class<T> clazz) throws ParsingException;
+
+    /**
+     * Returns the value of the specified (top level) key in the
+     * underlying JSON message as a child {@code ChangeEventElement}.
+     *
+     * @return optional {@code ChangeEventElement}, which is empty if
+     * the specified key is not found or if the value is null.
+     * @throws ParsingException if the underlying JSON message, or any
+     *                          of its parent messages are unparsable
+     */
+    Optional<ChangeEventElement> getChild(String key) throws ParsingException;
```

Comment:
keep in mind when you return `Optional`, it's creating more garbage

Reply:
I know, but I wanted to keep ParsingException for actual parsing errors... maybe use `NoSuchElementException` instead of some `Optional`s + some `hasXXX` where absolutely necessary? will give it a try

Reply:
All `Optional`s are gone now, together with any parsing code of our own.